### PR TITLE
[WIP2] Correction lastname

### PR DESCRIPTION
**Pull request by QuentinFourreaux**

### CODE REVIEW

**Comment by ptitloup** [comment]
malheureusement le code proposé ne respect pas les normes pep8. Voici ce qu'il faudrait corriger : 
./pod/authentication/populatedCASbackend.py:223:63: W291 trailing whitespace
./pod/authentication/populatedCASbackend.py:224:1: W191 indentation contains tabs
./pod/authentication/populatedCASbackend.py:224:1: E101 indentation contains mixed spaces and tabs
./pod/authentication/populatedCASbackend.py:224:3: E131 continuation line unaligned for hanging indent
./pod/authentication/populatedCASbackend.py:228:10: W291 trailing whitespace
./pod/authentication/populatedCASbackend.py:229:1: W191 indentation contains tabs
./pod/authentication/populatedCASbackend.py:229:1: E101 indentation contains mixed spaces and tabs
./pod/authentication/populatedCASbackend.py:229:3: E131 continuation line unaligned for hanging indent
./pod/authentication/populatedCASbackend.py:229:65: W291 trailing whitespace
./pod/authentication/populatedCASbackend.py:230:1: W191 indentation contains tabs
./pod/authentication/populatedCASbackend.py:230:1: E101 indentation contains mixed spaces and tabs
./pod/authentication/populatedCASbackend.py:230:4: E131 continuation line unaligned for hanging indent
./pod/authentication/populatedCASbackend.py:231:13: W191 indentation contains tabs
./pod/authentication/populatedCASbackend.py:231:13: E101 indentation contains mixed spaces and tabs
./pod/authentication/populatedCASbackend.py:231:14: E126 continuation line over-indented for hanging indent
./pod/authentication/populatedCASbackend.py:232:13: W191 indentation contains tabs
./pod/authentication/populatedCASbackend.py:232:13: E101 indentation contains mixed spaces and tabs
./pod/authentication/populatedCASbackend.py:233:13: W191 indentation contains tabs
./pod/authentication/populatedCASbackend.py:233:13: E101 indentation contains mixed spaces and tabs
./pod/authentication/populatedCASbackend.py:234:13: E121 continuation line under-indented for hanging indent
./pod/authentication/populatedCASbackend.py:234:14: W291 trailing whitespace
./pod/authentication/populatedCASbackend.py:235:1: W191 indentation contains tabs
./pod/authentication/populatedCASbackend.py:235:1: E101 indentation contains mixed spaces and tabs
./pod/authentication/populatedCASbackend.py:235:4: E131 continuation line unaligned for hanging indent

**Review by ptitloup — CHANGES_REQUESTED** [review]

File: pod/authentication/populatedCASbackend.py [hunk 1]
```diff
@@ -220,12 +220,19 @@ def populate_user_from_entry(user, owner, entry):
         else ""
     )
     user.last_name = (
-        entry[USER_LDAP_MAPPING_ATTRIBUTES["last_name"]].value
-        if (
-            USER_LDAP_MAPPING_ATTRIBUTES.get("last_name")
-            and entry[USER_LDAP_MAPPING_ATTRIBUTES["last_name"]]
-        )
-        else ""
+        entry[USER_LDAP_MAPPING_ATTRIBUTES['last_name']].value
```
[the remark this answers is not in the document]
Comment (IsabelleEven) [reply pod/authentication/populatedCASbackend.py:223]: En effet, ce serait plus simple, j'avais essayé autre chose mais pas cette écriture.

[the remark this answers is not in the document]
Comment (ptitloup) [reply pod/authentication/populatedCASbackend.py:223]: ok vous nous tenez informé ? Merci

[the remark this answers is not in the document]
Comment (IsabelleEven) [reply pod/authentication/populatedCASbackend.py:223]: Oui, pas de soucis, désolés pour la réponse tardive, on va regarder demain pour refaire un pull request.
On vous tient au courant.

**Comment by QuentinFourreaux** [comment]
Bonjour, 

Nous avons fait les commits aujourd'hui et proposé le pull request. 
Est-ce que tout est bon ? 

Merci 

Quentin FOURREAUX 

02.33.80.51.41 (15141) 
[ ***@***.*** | ***@***.*** ] 
Sauf le mardi: 8h-12h15 / 14h-17h45 
Mardi: 8h-13h 

Service Informatique 
02.33.80.85.10 (18510) 
[ ***@***.*** | ***@***.*** ] 
IUT Grand Ouest Normandie - Université de Caen 
Pôle universitaire d'Alençon 
61250 DAMIGNY 
IUT bureau A110 - L-V : 7h45-12h30 / 13h-18h 


De: "Ptitloup" ***@***.***> 
À: "EsupPortail/Esup-Pod" ***@***.***> 
Cc: "Quentin Fourreaux" ***@***.***>, "Author" ***@***.***> 
Envoyé: Jeudi 16 Juin 2022 15:02:41 
Objet: Re: [EsupPortail/Esup-Pod] [WIP2] Correction lastname (PR #628) 





@ptitloup commented on this pull request. 


In [ https://github.com/EsupPortail/Esup-Pod/pull/628#discussion_r899058796 | pod/authentication/populatedCASbackend.py ] : 
> @@ -220,12 +220,19 @@ def populate_user_from_entry(user, owner, entry):

         else ""

     )

     user.last_name = (

-        entry[USER_LDAP_MAPPING_ATTRIBUTES["last_name"]].value

-        if (

-            USER_LDAP_MAPPING_ATTRIBUTES.get("last_name")

-            and entry[USER_LDAP_MAPPING_ATTRIBUTES["last_name"]]

-        )

-        else ""

+        entry[USER_LDAP_MAPPING_ATTRIBUTES['last_name']].value 


ok vous nous tenez informé ? Merci

**Review by ptitloup — APPROVED** [review]

**Comment by ptitloup** [comment]
Totu est ok maintenant, on peut merger

**Pull request merged**